### DISCUSSION TITLE
JS - Operators - Ternary Operator

### DISCUSSION THREAD

**Discussion by mownica-webdev** [mownica-webdev/frontend_2024#22] [Tasks]
**1. Check if a number is odd or even.**

- Checks if input number is even or odd by using the modulus % operator.
- Print “Number is even” if the number is divisible by 2.
- Else print “Number is odd” if the number returns a remainder when divided by 2.

**2. Find the largest between two numbers using the > and = operators.**

- Print num1 is the largest if num1>num2.
- Print num2 is the largest if num1<num2.
- Else print num1 and num2 are equal when num1==num2.

**3. Find if the given number is within the provided start and end range using >=, <=, and && operators.**

- Print “Between the range” if num is between start and end values
- Else Print “Outside the range” since num is outside start and end values.

**Comment by RAKESHSARIPINENI9490** [comment]
```
var num = 11; 
if (num % 2 == 0) {
    console.log("Number is even");
} else {
    console.log("Number is odd");
}


var num1 = 10;
var num2 = 20;
if (num1 > num2) {
    console.log("num1 is the largest");
} else if (num1 < num2) {
    console.log("num2 is the largest");
} else {
    console.log("num1 and num2 are equal");
}

var num = 15;
var start = 10;
var end = 20;
if (num >= start && num <= end) {
    console.log("Between the range");
} else {
    console.log("Outside the range");
}

```
![Ternaryoperators](https://github.com/user-attachments/assets/c53d79d3-c536-4149-a484-e0ccddf50fee)

**Comment by nithin362** [comment]
const a=20;
const b=10;
console.log(a+b);

// ternary operator
let number = 10;
let result = (number % 2 === 0) ? "Even" : "Odd";
console.log(result);
// ternary operator
let age = 18;
let canVote = (age >= 18) ? "Yes, you can vote" : "No, you cannot vote";
console.log(canVote);

// largest number

let num1 = 10;
let num2 = 20;

let largest = (num1 > num2) ? `${num1} is the largest` :
              (num1 < num2) ? `${num2} is the largest` :
              `${num1} and ${num2} are equal`;

console.log(largest);

//range 

let num = 15;
let start = 10;
let end = 20;

let rangeCheck = (num >= start && num <= end) ? "Between the range" : "Outside the range";

console.log(rangeCheck);
![Screenshot 2024-09-14 103512](https://github.com/user-attachments/assets/a8c32c54-bf57-498a-b988-3560bf5bde18)

**Comment by Bhanusriboga** [comment]
var number1 = 10;
var number2 = 20; 
console.log(
    number1 > number2 ? "number1 is the largest" : number1 < number2 ? "number2 is the largest" : "number1 and number2 are equal"
);

const num=6;
const result = (num % 2 === 0) ? "Num is even" : "Num is odd";

console.log(result);

let num1 = 20;
let start = 10;
let end = 30;

let range = (num1 >= start && num1 <= end) ? "Between the range" : "Outside the range";

console.log(range);
![image](https://github.com/user-attachments/assets/8a332a57-d26a-4dc0-9be2-6b9b6ca550d8)

**Comment by Praneeth-Reddy66** [comment]
```
const a = 10 ;
const result = (a % 2 ==0 )
? "given number is even." : "given number is oss.";
console.log(result);

let num1 = 10;
let num2 = 20;
let z = (num1 > num2) ?"num1 is largest" : "num2 is larger";
console.log(z);

let number = 25;
let start = 10;
let end = 30;
let result1= (number >= start && number <= end )
? "true" :"false";
console.log(result1);
```
![Screenshot (33)](https://github.com/user-attachments/assets/e9e6bed9-e23f-4d9c-933b-7da44d989243)

**Comment by Jayachandra-git** [comment]
let number=24;
const result=(number%2 ===0 )? "EVEN ":"ODD";
console.log(result);

let num1=20;
let num2=15;
const res=(num1>num2)?"num1 is greater":"BOTH are equal";
console.log(res);

let num=25;
let start=20;
let end=30;
const res1=(num>=start && num<=end)?"Between the range":"out of the Range";
console.log(res1);

output...!!!
<img width="443" alt="Screenshot 2024-09-14 114838" src="https://github.com/user-attachments/assets/4c2fd845-4e75-423c-80ad-0e7936cc28b8">

**Comment by BhaskarVelpula** [comment]
```
const a=28;
const result= a%2==0 ?"it is odd number": "it is even number";
console.log(result);
//2. Find the largest between two numbers using the > and = operators.Print num1 is the largest if num1>num2.
//Print num2 is the largest if num1<num2.
//Else print num1 and num2 are equal when num1==num2.
let num1=28;
let num2=98;
let largestnumber= num1>num2 ?"largest number is num1":"largest number is num2";
console.log(largestnumber);
//3. Find if the given number is within the provided start and end range using >=, <=, and && operators.

//Print “Between the range” if num is between start and end values
//Else Print “Outside the range” since num is outside start and end values.
let d=30;
let start=0;
let end=29;
let output=(d>=start && d<=end)?"number in between range" :"number not in between range";
console.log(output);
```

![image](https://github.com/user-attachments/assets/78757d1f-aa8c-451c-bd07-b744c1450279)

**Comment by Vasavi2026** [comment]
```
const num1=25;
const output=(num1%2===0)? "Given number is even": "Given number is Odd";
console.log(output);

let A=20;
let B=35;
let C=(A>B)? "A is Greater": "B is Greater";
console.log(C);

let X=18;
let start=10;
let end=20;
let result=(X>=start && X<=end)? "Between the range": "Out of the range";
console.log(result);

```

![image](https://github.com/user-attachments/assets/dfcb1f91-89ec-447c-9b1f-be851899b7de)

**Comment by Srikanth268** [comment]
const number=18;
const result = (number % 2 === 0) ? "Number is even" : "Number is odd";

console.log(result);

const n1=12;
const n2=20;
console.log( n1>n2?"n1 largest number":n1<n2?"n2 largest number": "n1 equal to n2")

const value=40;
const start=10;
const end=30;
console.log(value>=start&&value<end?"Between the range":"outside the 
![Screenshot 2024-09-14 205158](https://github.com/user-attachments/assets/e2ad3acb-71fc-44fb-ac8d-c3253e1e1826)
range")``

**Comment by lakshmivuyyuru11** [comment]
```
//even and odd
let x=10
let y=x%2==0 ? "even nbr":"odd nbr";
console.log(y);
//largest
let num1=30;
let num2=20;
let num3=(num1>num2) ?"num1 is largest" : "num2 is largest";
console.log(num3);
//range
let num=10;
let start=20;
let end=30;
let range = (num >= start && num <= end) ? "Between the range" : "Outside the range";

console.log(range);

```
![task22 answer](https://github.com/user-attachments/assets/62c03bfb-49f7-408f-9ff9-82ccccd981a2)

**Comment by AkhilaPamula** [comment]
```
let a = 100;
let b = a % 2 == 0 ? "Given Num is Even" : "Given Num is Odd";
console.log(b);

var num1 = 100;
var num2 = 200;
var result = num1 > num2 ? "Num1 is Greater" : "Num2 is Greater";
console.log(result);

let num = 20;
let start = 30;
let end = 10;
var result = (num >= start && num <= end) ? "Between the range" : "Outside the range";
console.log(result);

```

![image](https://github.com/user-attachments/assets/4fa3260f-b165-444e-b34b-f86f088c2f95)

**Comment by ramyamarlapati27** [comment]
```
let length = 50;
let breadth = 40;
if (length===breadth){
    console.log("it is square");
}else{
    console.log("it is not square");
}

var num = 13; 
if (num % 2 == 0) {
    console.log("Number is even");
} else {
    console.log("Number is odd");
}




var num1 = 27;
var num2 = 20;
if (num1 > num2) {
    console.log("num1 is the largest");
} else if (num1 < num2) {
    console.log("num2 is the largest");
} else {
    console.log("num1 and num2 are equal");
}

var num = 15;
var start = 10;
var end = 20;
if (num >= start && num <= end) {
    console.log("Between the range");
} else {
    console.log("Outside the range");
}
```

![image](https://github.com/user-attachments/assets/1d31c072-7361-47c1-b095-30c1cb15d879)

**Comment by RoshithaKommineni** [comment]
```
let a=7;
let resul=(a%2==0)?"number is even":"number is odd";
console.log(resul);

let b=6;
let c=8;
if(b>c)
{
    console.log(b,"is greatest");
    
}else if(b<c){
    console.log(c,"is greatest");
}else{
    console.log("Both are equal");
    
}

let num=75;
let f=70;
let g=80;
let resulte=(num>=f && num<=g)?"Between Range":"Out Of Range"
console.log(resulte);

```
![image](https://github.com/user-attachments/assets/4bd0fac4-2274-4bfd-b4c9-86cc1c3c8bc0)

**Comment by Shravanivootla** [comment]
```
let a = 13;
let b = a % 2 == 0 ? " Num is Even" : " Num is Odd";
console.log(b);
 
var a1 = 10;
var b1 = 20;
var result = a1 > b1 ? "a1 is Greater" : "b1 is Greater";
console.log(result);

let n = 20;
let start = 30;
let end = 10;
var result = (n >= start && n <= end) ? "Between the range" : "Outside the range";
console.log(result);
```
![image](https://github.com/user-attachments/assets/7a927ea1-cd62-4477-886c-8d3cad8c2a1a)

**Comment by NagarjunaReddy1621** [comment]
```
 ///even or odd by using ternery///
 let x=24;
 const result=(x%2==0)? "it is a even": "it is a odd";
 console.log(result);
 ///check greater or not///
 let a=20;
 let b=10
 const output =(a>b)? "a is a greater" : "b is a greater";
 console.log(output);
///Range///
let num=25;
let start=20;
let end=30;
const rest=(num>=start && num<=end)?"Between the range":"out of the Range";
console.log(rest);
```
![Screenshot (31)](https://github.com/user-attachments/assets/58089824-34bf-45ff-a069-27fccf3766b1)

**Comment by bandichakravarthi** [comment]
```
var num = 10;
    if (num % 2 == 0) {
      console.log("Number is even");
    } else {
      console.log("Number is odd");
    }

    var num1=5;
    var num2=12;
        if (num1 > num2) {
          console.log(num1 + " is the largest");
        } else if (num1 < num2) {
          console.log(num2 + " is the largest");
        } else   
       {
          console.log(num1 + " and " + num2 + " are equal");   
      
        }
      
      
      var num=25;
      var start=10;
      var end=20;
     
        if (num >= start && num <= end) {
          console.log("Between the range");
        } else {
          console.log("Outside the range");
        }
      
  
  
```
![image](https://github.com/user-attachments/assets/c063f7b4-71aa-43bf-bdc2-93f4f593e2db)

**Comment by kirankittu718** [comment]
```
//even or odd
const x=18;
let result= x%2==0 ?"it is even number": "it is odd number";
console.log(result);

//greatest number
let no1=18;
let no2=7;
let largestnumber= no1>no2 ?"largest number is first number":"largest number is second number";
console.log(largestnumber);

//range
let n=18;
let s=7;
let e=45;
let output=(n>=s && n<=e)?"number is in range" :"number is not in range";
console.log(output);
```
![image](https://github.com/user-attachments/assets/a9e8343f-7808-465c-9e19-036fc2032696)

**Comment by SandhyaMeduri** [comment]
```
1.
// let number = 7; 

// let result = (number % 2 === 0) ? "Even" : "Odd";

// console.log(result);


2.
// let num1 = 10;
// let num2 = 25;

// let largest = (num1 > num2) ? num1 : num2;

// console.log(largest);



3.
let number = 15;   // The number to check
let start = 10;    // Start of the range
let end = 20;      // End of the range

let isInRange = (number >= start && number <= end) ? "Within Range" : "Out of Range";

console.log(isInRange);

        
```

![image](https://github.com/user-attachments/assets/377a5dde-5b36-4106-b05a-499f66eee384)
